providers/aws/transit_gateway.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"log"
+
+	"github.com/GoogleCloudPlatform/terraformer/terraform_utils"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+)
+
+var tgwAllowEmptyValues = []string{"tags."}
+
+type TransitGatewayGenerator struct {
+	AWSService
+}
+
+func (g *TransitGatewayGenerator) getTransitGateways(svc *ec2.Client) error {
+
+	p := ec2.NewDescribeTransitGatewaysPaginator(svc.DescribeTransitGatewaysRequest(&ec2.DescribeTransitGatewaysInput{}))
+	for p.Next(context.Background()) {
+		for _, tgw := range p.CurrentPage().TransitGateways {
+			g.Resources = append(g.Resources, terraform_utils.NewSimpleResource(
+				aws.StringValue(tgw.TransitGatewayId),
+				aws.StringValue(tgw.TransitGatewayId),
+				"aws_ec2_transit_gateway",
+				"aws",
+				tgwAllowEmptyValues,
+			))
+		}
+	}
+	return p.Err()
+}
+
+func (g *TransitGatewayGenerator) getTransitGatewayRouteTables(svc *ec2.Client) error {
+
+	p := ec2.NewDescribeTransitGatewayRouteTablesPaginator(svc.DescribeTransitGatewayRouteTablesRequest(&ec2.DescribeTransitGatewayRouteTablesInput{}))
+	for p.Next(context.Background()) {
+		for _, tgwrt := range p.CurrentPage().TransitGatewayRouteTables {
+			g.Resources = append(g.Resources, terraform_utils.NewSimpleResource(
+				aws.StringValue(tgwrt.TransitGatewayRouteTableId),
+				aws.StringValue(tgwrt.TransitGatewayRouteTableId),
+				"aws_ec2_transit_gateway_route_table",
+				"aws",
+				tgwAllowEmptyValues,
+			))
+		}
+
+	}
+	return p.Err()
+}
+
+func (g *TransitGatewayGenerator) getTransitGatewayAttachments(svc *ec2.Client) error {
+	p := ec2.NewDescribeTransitGatewayAttachmentsPaginator(svc.DescribeTransitGatewayAttachmentsRequest(&ec2.DescribeTransitGatewayAttachmentsInput{}))
+	for p.Next(context.Background()) {
+		for _, tgwa := range p.CurrentPage().TransitGatewayAttachments {
+			switch tgwa.ResourceType {
+			// terraform@v0.12.18 doesn't support this resource type
```

Comment:
ResourceType it's AWS SDK type, why terraform relevant for this?

Reply:
This was the error trace I got once I added the `aws_ec2_transit_gateway_vpn_attachment` resource:

```
2020/01/13 13:25:29 aws importing... transit_gateway
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_route_table.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpn_attachment.tfer--xxx
panic: unknown resource type aws_ec2_transit_gateway_vpn_attachment

goroutine 536 [running]:
github.com/hashicorp/terraform/plugin.(*GRPCProvider).getResourceSchema(0xc0000c60e0, 0x3450d70, 0x26, 0x0, 0x0)
        /Users/dbarranco/go/pkg/mod/github.com/hashicorp/terraform@v0.12.18/plugin/grpc_provider.go:92 +0x129
github.com/hashicorp/terraform/plugin.(*GRPCProvider).ReadResource(0xc0000c60e0, 0x3450d70, 0x26, 0x3939400, 0xc0006f8180, 0x2e087e0, 0xc001322d20, 0x5449c30, 0x0, 0x0, ...)
        /Users/dbarranco/go/pkg/mod/github.com/hashicorp/terraform@v0.12.18/plugin/grpc_provider.go:321 +0xa3
github.com/GoogleCloudPlatform/terraformer/terraform_utils/provider_wrapper.(*ProviderWrapper).Refresh(0xc0004463c0, 0xc000447310, 0xc0004472c0, 0x2, 0xc00058a2a0, 0x68)
```

Doesn't it seems to be related with the version of Terraform that the project is currently using?

Reply:
I think it's related your aws provider version.

Reply:
`aws_ec2_transit_gateway_vpn_attachment ` is a data source, not a resource.

Reply:
crap, it is true.
The only current Terraform resource supported at the moment are the vpc attachments.
Will change that.